### PR TITLE
fix: make second RTCPeerConnection constructor parameter optional  (#32)

### PR DESCRIPTION
**Pull request by athouary**

### CODE REVIEW

**Comment by saghul** [comment]
@andrei-gavrilescu can you PTAL?

File: rtcstats.js [hunk 1]
```diff
@@ -172,6 +172,7 @@ export default function(
             // does not affect the normal flow of any application that might integrate it.
             const origConfig = { ...config };
             const origConstraints = { ...constraints };
+            constraints = constraints || {};
```
Comment (andrei-gavrilescu) [inline rtcstats.js:175]: would giving constraints a default parameter value `constraints = {}` work as well? maybe it's a bit cleaner.

File: rtcstats.js [hunk 2]
```diff
@@ -214,8 +215,8 @@ export default function(
                 // TODO: do we want to log constraints here? They are chrome-proprietary.
                 // eslint-disable-next-line max-len
                 // http://stackoverflow.com/questions/31003928/what-do-each-of-these-experimental-goog-rtcpeerconnectionconstraints-do
-                if (constraints) {
-                    sendStatsEntry('constraints', id, constraints);
+                if (origConstraints) {
+                    sendStatsEntry('constraints', id, origConstraints);
```
Comment (andrei-gavrilescu) [inline rtcstats.js:219]: if we give constraints a default value there's no point in checking if it's undefined, we can simply sendStatsEntry in all cases.

**Comment by andrei-gavrilescu** [comment]
Is this change required in the associated ljm PR? if so then rtcstats would need to be updated there as well.

File: rtcstats.js (hunk 2 above)
Comment (athouary) [reply rtcstats.js:219]: I assumed we would not want to `sendStatsEntry('constraints', ...)` in all cases, that's why I did not use the default parameter. That way, I could still know whether the consumer passed constraints or not.

However, if you think it's ok to call `sendStatsEntry('constraints', ...)` regardless of the constraints object containing anything, then I'll be happy to add a default to the parameter declaration.

Comment (athouary) [reply rtcstats.js:219]: @andrei-gavrilescu Is it acceptable to send stats entries for constraints when there are none?

Comment (andrei-gavrilescu) [reply rtcstats.js:219]: Apologies for taking this long to reply, got a bit sidetracked.
I think because we use the construct `const origConstraints = { ...constraints };`,  `origConstraints` will always have a value event if `constraints` is undefined so the if will pass regardless, removing it will just send an empty object, in rtcstats-server we can just assume that no constraints were passed if that is the case.
TLDR, yes it is acceptable, feel free to remove the if.

**Comment by andrei-gavrilescu** [comment]
> This change is not directly related to the mentioned ljm PR. However, rtcstats does need to be updated there. Should I open a PR for that in ljm?

Got it, I'll make npm release of rtcstats after this gets merged. I'll ping you with the new version. 
Afterwards if you're willing to bump the version in ljm that'd be great, if you don't have the time I can just bump it later on.

**Comment by athouary** [comment]
@andrei-gavrilescu  Just a little reminder about making an npm release. I'll take care of bumping the version in ljm.

Thanks!

**Comment by andrei-gavrilescu** [comment]
> @andrei-gavrilescu Just a little reminder about making an npm release. I'll take care of bumping the version in ljm.
> 
> Thanks!

Thanks for the reminder, v9.7.0 has been published to npm.
